browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -75,17 +151,24 @@ void OnBeforeURLRequestAdBlockTP(
   }
   DCHECK_NE(ctx->request_identifier, 0UL);
 
-  g_brave_browser_process->ad_block_service()->GetTaskRunner()
-      ->PostTaskAndReply(FROM_HERE,
-                         base::BindOnce(&ShouldBlockAdOnTaskRunner, ctx),
-                         base::BindOnce(&OnShouldBlockAdResult, next_callback,
-                                        ctx));
+  auto* rph = content::RenderProcessHost::FromID(ctx->render_process_id);
+  if (!rph) {
+    // This case occurs for the top-level request. It's okay to ignore the
```

Comment:
top-level requests don't have a render process host? That doesn't sound right and also if top-level requests can't be blocked, why are we calling `ShouldBlockAdOnTaskRunner`?

Reply:
cc @iefremov

Reply:
`ctx->render_process_id` comes up as -1 for top level requests from my testing, which causes the RenderProcessHost to be a `nullptr`.

Reply:
resolved by getting required fields through `GetWebContents`